lib/scripts/collection_list.rb

```diff
@@ -0,0 +1,32 @@
+def collections_path(collection)
+  if collection.collections.empty?
+    collection_paths = [[{"title": collection.title, "id": collection.id}]]
+  else
+    collection_paths = []
+  end
+
+  collection.collections.each do |parent_collection|
+    recursive_collection_path(
+      parent_collection,
+      [{"title": parent_collection.title, "id": parent_collection.id},
+       {"title": collection.title, "id": collection.id}],
+      collection_paths
+    )
+  end
+
+  collection_paths
+end
+
+def recursive_collection_path(collection, path, collection_paths)
+  if collection.collections.empty?
+    collection_paths << path
+  else
+    collection.collections.each do |collection|
+      recursive_collection_path(
+        collection,
+        [{"title": collection.title, "id": collection.id}] + path,
```

Comment:
`{"title": collection.title, "id": collection.id}` shows up in both `#recursive_collection_path` and `#collections_path`. Also, `{"title": parent_collection.title, "id": parent_collection.id}` is basically the same idea, but for a parent collection. It should be wrapped into a function that describes what this array is:

```
def name_describing_data(collection)
  [{"title": collection.title, "id": collection.id}]
end
```